### ISSUE TITLE
access_token to short but passing authentication

### ISSUE THREAD

**Issue by wpitallo** [auth0/auth0.js#670]
Hi I have been using the new version 9.2.2 for about a two weeks, everything was working perfectly. Yesterday however I created a new user logged in everything was fine, then all of a sudden I started getting unauthorized results back from my server. I investigated and found that my the access_token was way to short about the size of a regular guid. I logged out and back in and everything was ok again but this has me quite concerned. What caused the short access token to be returned? Each time I refreshed the page auth0.js also said that I was authenticated even with this short access token. This should not be happing, also worring that its easy to reproduce? Any ideas what's wrong here?

**Comment by lbalmaceda** [comment]
Access Tokens are opaque and the format or length shouldn't matter to you. You can always check if a token is valid calling the [/userinfo endpoint](https://auth0.com/docs/api/authentication#get-user-info). If you got a 401 is probably because the token had expired. You may be able to find more details in the *Logs* section of the dashboard. Note that in the login response along with all the tokens you get the `expires_in` value telling you how many seconds you have left before the tokens expire.

**Comment by wpitallo** [comment]
The token had not expired, if so it normally redirects from the client page to the login screen. It was returning as authenticated from auth0.js but server side throwing an error. The only difference between that login  and every other one was the length of the token every single other time I log successfully the token is much longer. Are the longer not tokens JWT tokens carrying claims etc needed for server side authentication?

**Comment by luisrudge** [comment]
When you say server side, you mean your server or auth0's server? As @lbalmaceda said, access tokens are opaque and have no meaning at all. They can be GUIDs, JWTs, or anything else, so you shouldn't rely on it being anything, it's just a string. id tokens, however, are guaranteed to be JWTs.

In any case, this isn't an SDK issue, so please get in touch with our amazing support team at https://support.auth0.com 🎉

**Comment by wpitallo** [comment]
Thank you!

**Comment by wpitallo** [comment]
@luisrudge so I tried using the id_token as the bearer header and it does not work! I followed this tutorial: https://auth0.com/docs/quickstart/backend/webapi-owin/01-authorization. When I pass in the access_token it authorizes server side but only when it is the correct length. I have still not reproduced this again but there was a shot token being returned and it did not authorize on the server side but did on the client. So why does the id_token not authorize on server side but the access_token does? Also have a look at this tutorial here it is the access_token that is being passed to the server. https://auth0.com/blog/developing-web-apps-with-asp-dot-net-core-2-dot-0-and-react-part-2/.

So in summary the first tutorial states that the its using JWT, Microsoft.Owin.Security.Jwt. and in the second tutorial it clearly states that the access_token should be used: 

const accessToken = localStorage.getItem('access_token');

This makes sense and back to my point therefore this will only work if the access_token is also a JWT. and it will fail if it is not?

**Comment by luisrudge** [comment]
it all depends on the audience param: https://auth0.com/docs/tokens/access-token#access-token-format

**Comment by wpitallo** [comment]
Yes and my Audience is set to my unique identifier of a custom API.  'https://xxxx' so it should then always be returning a JWT. So this must then be a bug, why would it once in two weeks (probably hundreds of logins) only once return a guid instead of a JWT?

**Comment by luisrudge** [comment]
The best thing to do here is contact our amazing support team at https://support.auth0.com and send [HAR](https://auth0.com/docs/support/troubleshooting-with-har-files) files to them with the requests that you get access_token as JWT/opaque string. They can analyze the HAR files and give you a better answer on what's going on.

**Comment by wpitallo** [comment]
Ok cool thanks will do :)

**Comment by wpitallo** [comment]
So it occurred again, 12 days later. I tried logging it on the community form but that was horrible experience and I eventually gave up as I kept getting, "you do not have permission to access that resource" every time i tried to submit :(. I eventually sent a mail to sales. :(

access_token
:"MeidrDhARt0aJbgyeejPDhU1YAexxxxxxx"

**Comment by jerdog** [comment]
@wpitallo - we just switched platforms and so I am looking to identify what issue you're currently having with submitting in the forum. Feel free to contact me directly at jeremy.meiss@auth0.com and I will figure it out
